packages/adp-tooling/src/base/change-utils.ts

```diff
@@ -0,0 +1,202 @@
+ *
+ * @param {string} propertyValue - The property value to be parsed.
+ * @returns {PopertyValueType} The parsed value if `propertyValue` is valid JSON; otherwise, returns the original `propertyValue`.
+ * @example
+ * // Returns the object { key: "value" }
+ * getParsedPropertyValue('{"key": "value"}');
+ *
+ * // Returns the string "nonJSONValue" because it cannot be parsed as JSON
+ * getParsedPropertyValue('nonJSONValue');
+ */
+export function getParsedPropertyValue(propertyValue: PropertyValueType): PropertyValueType {
+    try {
+        const value = JSON.parse(propertyValue);
+        return value;
+    } catch (e) {
+        return propertyValue;
+    }
+}
+
+/**
+ * Searches for a change file with a specific inbound ID within a project's change directory.
+ *
+ * @param {string} projectPath - The root path of the project.
+ * @param {string} inboundId - The inbound ID to search for within change files.
+ * @returns {InboundChangeData} An object containing the file path and the change object with the matching inbound ID.
+ * @throws {Error} Throws an error if the change file cannot be read or if there's an issue accessing the directory.
+ */
+export function findChangeWithInboundId(projectPath: string, inboundId: string): InboundChangeData {
+    let changeObj: InboundChange | undefined;
+    let filePath = '';
+
+    const pathToInboundChangeFiles = path.join(projectPath, '/webapp/changes/manifest');
+
+    if (!existsSync(pathToInboundChangeFiles)) {
+        return {
+            filePath,
+            changeWithInboundId: changeObj
+        };
+    }
+
+    try {
+        const file: Dirent | undefined = readdirSync(pathToInboundChangeFiles, { withFileTypes: true }).find(
+            (dirent: Dirent) => dirent.isFile() && dirent.name.includes('changeInbound')
+        );
+
+        if (file) {
+            const pathToFile = path.join(pathToInboundChangeFiles, file.name);
+            const change: InboundChange = JSON.parse(readFileSync(pathToFile, 'utf-8'));
+
+            if (change.content?.inboundId === inboundId) {
+                changeObj = change;
+                filePath = pathToFile;
+            }
+        }
+
+        return {
+            filePath,
+            changeWithInboundId: changeObj
+        };
+    } catch (e) {
+        throw new Error(`Could not find change with inbound id '${inboundId}'. Reason: ${e.message}`);
+    }
+}
+
+/**
+ * Constructs a generic change object based on provided parameters.
+ *
+ * @param {T} data - The base data associated with the change, including project data and timestamp.
+ * @param {object} content - The content of the change to be applied.
+ * @param {GeneratorName} generatorName - The name of the generator creating this change.
+ * @param {ChangeTypes} changeType - The type of the change.
+ * @returns An object representing the change.
+ * @template T - A type parameter extending `BaseData`.
+ */
+export function getGenericChange<T extends BaseData>(
```

Comment:
Why do you use generics here? If you just make `date: BaseData` then you are good to go. I don't see any generic usage, or am I missing something?

Reply:
Changed in: [3ed676b](https://github.com/SAP/open-ux-tools/pull/1643/commits/3ed676b560a71c69eac49e8258cf84f78c948ade)

Reply:
Thanks, I think you need to update the JSDoc comments to reflect the change of the input parameters.

Reply:
[29624c3](https://github.com/SAP/open-ux-tools/pull/1643/commits/29624c3030aaf5a3a29179b8686642cd55790aa3)